### ISSUE TITLE
Can't get admin role

### ISSUE THREAD

**Issue by sham-sr** [swan-cern/jupyterhub-extensions#33]
I'm probably wrong somewhere, but I can't figure out where.
I set the role in config (KEYCLOAK_ADMIN_ROLE=admins )
```c.KeyCloakAuthenticator.admin_role = os.environ.get('KEYCLOAK_ADMIN_ROLE')``` 
in KeyCloak bind user to the role  admins
and  when i Authentication i get
```Authentication successful for user: admin, roles: {'e', 'n', 'o', 'r', 'l', '_', 's'}, admin: `False```

**Comment by diocas** [comment]
Hi @sham-sr ,
TL;DR: This means that your token does not contain any roles. 

When you set "admins" as the `admin` role, this means that the authenticator will look in your token and check if it contains a role with that name. But the message you got means that you have 0 roles so "admins" is not found in an empty list, ergo you're not an admin.

I suggest you check your token (you can use https://jwt.io/ , for example) to see if it's properly configured. This authenticator gets the roles list from `resource_access` > `<your-app-id>` > `roles`.
(I assume this is a standard place to have them. If not and you have them somewhere else, maybe we can make the authenticator configurable).

Cheers
Diogo

**Comment by sham-sr** [comment]
Hi Diogo,
Thanks for your help and your work. It all worked. The newcomer in Keycloak added steering wheels to realm, but did not add them to the client.
